app/app.go

```diff
@@ -368,7 +368,11 @@ func initICS4Wrapper(keepers *keeperCache, wasmHooks ibchooks.WasmHooks) ibchook
 		getKeeper[axelarnetKeeper.Keeper](keepers),
 	)
 	// create a middleware to integrate wasm hooks into the ibc pipeline
-	return ibchooks.NewICS4Middleware(ics4Wrapper, wasmHooks)
+	if wasmHooks.ProperlyConfigured() {
```

Comment:
this requires contract keeper to be not nil if ibc hooks was enabled, but we're setting that after calling this method. so ibc hooks won't work when enabled
```
	// set the contract keeper for the Ics20WasmHooks
	wasmHooks.ContractKeeper = getKeeper[wasmkeeper.PermissionedKeeper](keepers)
```

Reply:
good catch 🤦‍♂️